### PR TITLE
ORV2-1379 Update Presentation of PermitType Column in Applications Tables

### PR DESCRIPTION
**Pull request by erikataot**
# Description

- Provide more consistency between the 3 Applications tables
- Display the permit type acronym in the `PermitType` column (aka. `TROS`)
- Upon hover of the table cell, display a tooltip of the Permit Type's long version (aka. `Term Oversize`, `Term Overweight`)
- Updated Permit Search Results table to also include the Tooltip for Permit Type

Screenshot 2024-04-04 095350

Screenshot 2024-04-04 095424

image

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1306-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1306-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1306-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1306-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1306-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1306-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1306-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1306-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/features/permits/components/permit-list/ApplicationInProgressColumnDefinition.tsx [hunk 1]
```diff
@@ -42,6 +44,12 @@ export const ApplicationInProgressColumnDefinition = (
       id: "permitType",
       enableSorting: false,
       header: "Permit Type",
+      Cell: (props: {cell: any; row: any }) => {
```
Comment (krishnan-aot) [inline frontend/src/features/permits/components/permit-list/ApplicationInProgressColumnDefinition.tsx]: `row` can be removed.

**Review by krishnan-aot — COMMENTED** [review]
Is this change needed for the Permit Search component as well?

**Comment by erikataot** [comment]
> Is this change needed for the Permit Search component as well?

good question! We saw the desired acronym in the permit search results during our meeting, but to be consistent I'll follow up with them on the hovering of the permit type name.

File: frontend/src/features/permits/apiManager/permitsAPI.ts [hunk 2]
```diff
@@ -141,7 +140,7 @@ export const getApplicationsInProgress = async ({
         paginatedApplications.items.map((application) => {
           return {
             ...application,
-            permitType: getPermitTypeName(application.permitType) as string,
+            permitType: application?.permitType,
```
Comment (krishnan-aot) [inline frontend/src/features/permits/apiManager/permitsAPI.ts]:
```suggestion
```

**Comment by erikataot** [comment]
> > Is this change needed for the Permit Search component as well?
> 
> good question! We saw the desired acronym in the permit search results during our meeting, but to be consistent I'll follow up with them on the hovering of the permit type name.

Yup, for consistency I've added it to the Permit Search results as well. Good catch.

**Review by krishnan-aot — APPROVED** [review]

**Pull request merged**